tests/integration/test_s3.py

```diff
@@ -248,3 +248,25 @@ def test_s3_upload_download_gzip():
         downloaded_data = filestream.read().decode('utf-8')
 
     assert downloaded_data == data, '{} != {}'.format(downloaded_data, data)
+
+
+def test_bucket_exists():
+    # Test setup
+    bucket = 'test-bucket'
+
+    s3_client = aws_stack.connect_to_service('s3')
+    s3_client.create_bucket(Bucket=bucket)
+    s3_client.put_bucket_cors(
+        Bucket=bucket,
+        CORSConfiguration={
+            'CORSRules': [{'AllowedMethods': ['GET', 'POST', 'PUT', 'DELETE'],
+                           'AllowedOrigins': ['localhost']}]
+        }
+    )
+
+    response = s3_client.get_bucket_cors(Bucket=bucket)
+    print(str(response))
+    assert response['ResponseMetadata']['HTTPStatusCode'] == 200
```

Comment:
In addition to the positive case, could we also test the negative case (with non-existing bucket) and assert a 404 return code?